v7/navstories/navstories.py

```diff
@@ -38,12 +38,50 @@ def set_site(self, site):
         site.scan_posts()
         # NAVIGATION_LINKS is a TranslatableSetting, values is an actual dict
         for lang in site.config['NAVIGATION_LINKS'].values:
+            # Which paths are navstories active for for lang?
+            navstories_paths = ()
+            if 'NAVSTORIES_PATHS' in site.config and lang in site.config['NAVSTORIES_PATHS']:
+                navstories_paths = site.config['NAVSTORIES_PATHS'][lang]
+
             new = []
+            newsub = {}
             for p in site.pages:
-                if lang in p.translated_to and not p.meta('hidefromnav'):
-                    new.append(p)
-            new_entries = tuple(sorted([(p.permalink(lang), p.title(lang)) for p in new]))
-
+                permalink = p.permalink(lang)
+                for s in navstories_paths:
+                    if permalink.startswith('/' + s + '/'):
```

Comment:
Make the check using a tuple, then use a for loop to figure out the **longest** string that the permalink starts with.

Or perhaps:

```py
navstories_paths = tuple(('/' + s.strip('/') + '/') for s in navstories_paths)

for p in site.pages:
    permalink = p.permalink()
    s_candidates = [s for s in navstories_paths if permalink.startswith(s)]
    if not s_candidates:
        continue
    # get longest path
    s = max(s_candidates, key=len)

    navpath = permalink[len(s):].split('/')
```

Use case:

```py
NAVSTORIES_PATHS = {DEFAULT_LANG: ('pages', 'pages/foo', 'pages/foo/bar')}
```